packages/wasm-parser/src/decoder.js

```diff
@@ -959,6 +962,59 @@ export function decode(ab: ArrayBuffer, opts: DecoderOpts): Program {
     return t.globalType(type, globalType);
   }
 
+  // this section contains an array of function names and indices
+  function parseNameSectionFunctions() {
+    const functionNames = [];
+
+    const subSectionSizeInBytesu32 = readU32();
+    eatBytes(subSectionSizeInBytesu32.nextIndex);
+
+    const numbeOfFunctionsu32 = readU32();
+    const numbeOfFunctions = numbeOfFunctionsu32.value;
+    eatBytes(numbeOfFunctionsu32.nextIndex);
+
+    for (let i = 0; i < numbeOfFunctions; i++) {
+      const indexu32 = readU32();
+      const index = indexu32.value;
+      eatBytes(indexu32.nextIndex);
+
+      const name = readUTF8String();
+      eatBytes(name.nextIndex);
+
+      functionNames.push(t.functionNameMetadata(name.value, index));
+    }
+    return functionNames;
+  }
+
+  function parseNameSectionLocals() {
+    const subSectionSizeInBytesu32 = readU32();
+    const subSectionSizeInBytes = subSectionSizeInBytesu32.value;
+    eatBytes(subSectionSizeInBytesu32.nextIndex);
+
+    // TODO: add support for naming locals - for now we just eat this section
+    eatBytes(subSectionSizeInBytes);
+  }
+
+  // this is a custom suction that wat2wasm includes if invoked
+  // using the --debug-names option
+  function parseNameSection(remainingBytes: number) {
```

Comment:
I'm wondering if we shouldn't split it in a `wabt` file or something? Because it's specific to the binary they're emitting.

Also how we could know that it has been emitted by `wabt`?

Reply:
good point - my next step will be to complete this work by supporting locals, so I could have a go at that then?

Reply:
Yes sure